katharsis-core/src/main/java/io/katharsis/queryspec/DefaultQuerySpecDeserializer.java

```diff
@@ -192,7 +200,11 @@ private void deserializePage(QuerySpec querySpec, Parameter parameter) {
 			querySpec.setOffset(parameter.getLongValue());
 		}
 		else if (LIMIT_PARAMETER.equalsIgnoreCase(name)) {
-			querySpec.setLimit(parameter.getLongValue());
+			Long limit = parameter.getLongValue();
+			if(getMaxPageLimit() != null && limit != null) {
+				limit = Math.min(limit, getMaxPageLimit());
```

Comment:
I would throw here a BadRequestException or maybe 416 Requested range not satisfiable. Since the caller does not get what he requested, we should fail fast. Otherwise it may lead to follow up errors that go unnoticed.